### PR TITLE
[Go] Add support for the Mars web toolkit

### LINKED ISSUE #1957: Add mars
Hi @roblillack,

Would you mind to add (or consent that we add) [mars](https://github.com/roblillack/mars) in `here` ?

This project compare performance based on 3 routes :

+ `GET` `/`, **SHOULD** return a successful status code and an empty string
+ `GET` `/user/:id`, **SHOULD** return a successful status code and the id
+ `POST` `/user`, **SHOULD** return a successful status code and an empty string

Regards,

### PR DESCRIPTION
**Pull request by roblillack**
Hey there, did my best to add support for [mars](https://github.com/roblillack/mars) but I cannot seem to get `neph` to run on the machine, I'm currently at---so cannot fully test.

Ticket: https://github.com/the-benchmarker/web-frameworks/issues/1957

-----

## If you are adding a framework

+ [x] A `Dockerfile` exists, or can be built ?
+ [x] I've updated `CI` config file (and **commited** `.travis.yml`)
~~~sh
bin/make ci_config
~~~
+ [x] I've updated `.dependabot` config file  (and **commited** `.dependabot/config.yml`)
~~~sh
bin/make deps_config
~~~
+ [ ] All tests passes ?
~~~
export FRAMEWORK=<MY_FRAMEWORK>
shards install
shards build --release
bin/make config
bin/neph ${FRAMEWORK}
crystal spec
~~~

-------------
edit : Closes #1957

### CODE REVIEW

File: go/mars/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module main
+
+go 1.12
```
Comment (waghanza) [inline go/mars/go.mod]: could we remove this line ? 
+ I know go generate it, but we will not stick to a particular go version
+ We are already in `go` **1.13**

File: go/mars/routes.conf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+GET     /                           App.Index
+GET     /user/:id                   App.GetUser
+POST    /user                       App.PostUser
```
Comment (waghanza) [inline go/mars/routes.conf]: I have a **404** for the last route

File: go/mars/config.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+framework:
+  website: github.com/roblillack/mars
```
Comment (waghanza) [inline go/mars/config.yaml]: could you use shortcut 
```yaml
github: roblillack/mars
```

File: go/mars/go.mod (hunk 1 above)
Comment (roblillack) [reply go/mars/go.mod]: Actually that line specifies the minimal version of Go that this code will compile with. I removed it anyway.

File: go/mars/register_controllers.gen.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
+// DO NOT EDIT -- code generated by mars-gen
+package main
```
Comment (waghanza) [inline go/mars/register_controllers.gen.go]: could this file be generated on the container (before compilation) ?

Comment (roblillack) [reply go/mars/register_controllers.gen.go]: Yes, just add `go generate main.go` before running the build.

Comment (waghanza) [reply go/mars/register_controllers.gen.go]: may I push a fix on your branch ?

Comment (roblillack) [reply go/mars/register_controllers.gen.go]: Yes, sure!

**Comment by waghanza** [comment]
@roblillack Is this `PR` ready ?

**Comment by roblillack** [comment]
@waghanza, you decide. It looks ready from my side, but I'm not able to run the testsuite with neph. Feel free to merge anytime!

**Pull request merged**